### PR TITLE
This commit fixes issue #175 (be_empty matcher)

### PR DESCRIPTION
**Pull request by proldan**

### CODE REVIEW

**Comment by Pazus** [comment]
@proldan please check your changes, some tests are failling

File: source/expectations/data_values/ut_data_value_refcursor.tpb [hunk 1]
```diff
@@ -53,6 +53,24 @@ create or replace type body ut_data_value_refcursor as
...
-
+  
+  member function is_empty return boolean is  
+    l_is_empty boolean := FALSE;
+    l_result CLOB;
+  begin
+    if self.data_value is not null then
+      ut_assert_processor.set_xml_nls_params();
+      dbms_xmlgen.restartQuery(self.data_value);
+      dbms_xmlgen.setMaxRows(self.data_value, 100);
```
Comment (Pazus) [inline source/expectations/data_values/ut_data_value_refcursor.tpb]: I think maxrows can be set to 1

**Comment by jgebal** [comment]
The be_empty matcher should also work for collections (anydata). Can you add that so the matcher is complete?

**Comment by jgebal** [comment]
For anydata, if the type is object, the matcher should fail with error that the matcher cannot be used on objects but only on collections.

**Comment by proldan** [comment]
@Pazus  what tests failed? The ones on the be_empty matcher or other ones? I had some problems installing develop version on my db and scheman and I can't execute some of them...

**Comment by jgebal** [comment]
You can see the build for this PR failing 
"Some checks have failed"
Click on details and you see the build results

**Comment by proldan** [comment]
@jgebal Thanks! The matcher should fail with objects and any other type except sys_refcursor, right?

**Comment by jgebal** [comment]
The matches schuld succeed for empty dataset in ref cursor and for empty collecion passed to the matcher as anydata

**Comment by proldan** [comment]
Corrected! Travis CI build passed and I added support for collections

**Comment by Pazus** [comment]
Thanks, @proldan . Merged.

**Pull request merged**